weave/trace/annotation_parser.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+import logging
+import re
+from inspect import Signature
+
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+CONTENT_CLASS_NAME = "weave.type_wrappers.Content.content.Content"
+
+# Content annotation with a nested Literal type
+# Example: "typing.Annotated[str, <class 'my_app.Content[key=Literal['user_id']]'>]"
+# Explanation:
+# - r"typing\.Annotated\[": Matches the literal "typing.Annotated[".
+# - r"(.+?),\s*": Group 1. Captures the base type (e.g., "str") and the following comma.
+# - rf"<class '{re.escape(CONTENT_CLASS_NAME)}\[": Matches the start of the metadata class string.
+# - r"[^\]]*Literal\[['\"](.+?)['\"]\]": Matches any character except ']' until it finds a Literal, then captures its string content as Group 2.
+# - r"\]'>": Matches the closing characters for the metadata class representation (e.g., "]'>").
+# - r"\]": Matches the final closing bracket of the Annotated type.
+
+PATTERN_WITH_TYPE_HINT = re.compile(
+    r"typing\.Annotated\["
+    r"(.+?),\s*"  # Group 1: Base type (non-greedy)
+    rf"<class '{re.escape(CONTENT_CLASS_NAME)}\["
+    r"[^\]]*Literal\[['\"](.+?)['\"]\]"  # Group 2: Any literal, non-greedy
+    r"\]'>"  # Closing bracket for Content[...] and class
+    r"\]"  # Closing bracket for Annotated[...]
+)
```

Comment:
will this work e.g. if the user writes `Annotated[...]` instead of `typing.Annotated[...]`?

Reply:
fixed so it will match any library to import from.